MDCStatefulRipple.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCRippleView.h"
+
+/**
+ Provides the current state of the ripple. The ripple is either in its normal state, or in the
+ selected state where the ripple remains spread on the view.
+ - MDCRippleStateNormal: The ripple isn't currently presented.
+ - MDCRippleStateHighlighted: The ripple is activated and shown.
+ - MDCRippleStateSelected: The ripple is in the selected state.
+ - MDCRippleStateDragged: The ripple is in the dragged state.
+ */
+typedef NS_OPTIONS(NSInteger, MDCRippleStateNew) {
+  MDCRippleStateNewNormal = 0,
+  MDCRippleStateNewHighlighted = 1 << 0,
+  MDCRippleStateNewSelected = 1 << 1,
+  MDCRippleStateNewDragged = 1 << 2,
+};
```

Comment:
What about an "Active" state?  Tabs, for instance, have an active state ("active" might have a different theming than "selected").

Reply:
True. Unfortunately we don't yet have clear guidance on how the Activated state will behave in terms of overlays, so once we hash that out we can add the additional state and its behavior along side the other state.